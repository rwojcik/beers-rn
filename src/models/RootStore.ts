import { types, Instance, onSnapshot } from 'mobx-state-tree';
import { useContext, createContext } from 'react';
import { BeersStore } from './BeersStore';

export const RootModel = types.model('RootModel', {
    beers: BeersStore,
});

export const rootStore = RootModel.create({
    beers: { state: 'initial' },
});

export type RootInstance = Instance<typeof RootModel>;
export const RootStoreContext = createContext<null | RootInstance>(null);

export const RootStoreProvider = RootStoreContext.Provider;

export function useRootStore() {
    const store = useContext(RootStoreContext);
    if (store == null) {
        throw new Error('Store cannot be null, please add a context provider');
    }
    return store;
}
