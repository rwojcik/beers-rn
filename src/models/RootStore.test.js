import { rootStore } from './RootStore';

it('creates rootStore instance with beers store', () => {
    expect(rootStore.beers.state).toBe('initial');
    expect(rootStore.beers.data).toHaveLength(0);
});

it('creates rootStore instance and matches snapshot', () => {
    expect(rootStore).toMatchSnapshot();
});
