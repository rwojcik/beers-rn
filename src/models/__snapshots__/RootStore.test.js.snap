// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates rootStore instance and matches snapshot 1`] = `
Object {
  "beers": Object {
    "data": Array [],
    "hasMore": true,
    "page": 1,
    "state": "initial",
  },
}
`;
